json11.cpp

```diff
@@ -426,11 +419,14 @@ struct JsonParser final {
      */
     void consume_garbage() {
       consume_whitespace();
+      if (failed) return;
```

Comment:
You're right.  I guess I was following a safe pattern with separation-of-concerns.  But at this layer of implementation it might actually make sense to avoid extra checks for performance reasons.  I'll remove them.